Issue 342 aws iam role - closes #342 (#644)
bob-crutchley/courseware: topics/aws/modules/aws-iam-role/README.md

# AWS IAM Role

[aws-signin]: https://signin.aws.amazon.com/signin?redirect_uri=https%3A%2F%2Feu-west-1.console.aws.amazon.com%2Fconsole%2Fhome%3Fnc2%3Dh_ct%26region%3Deu-west-1%26src%3Dheader-signin%26state%3DhashArgs%2523%26isauthcode%3Dtrue&client_id=arn%3Aaws%3Aiam%3A%3A015428540659%3Auser%2Fhomepage&forceMobileApp=0

[running-instances]: https://eu-west-1.console.aws.amazon.com/ec2/v2/home?region=eu-west-1#Instances:sort=instanceId

[iam-service-search]: https://i.imgur.com/O3Ejc6f.png

[iam-create-role]: https://i.imgur.com/Iihk7sh.png

[iam-role-create-role]: https://i.imgur.com/m3LyIu7.png

[iam-role-rds-policy]: https://i.imgur.com/QmtE2bv.png

[iam-role-tag]: https://i.imgur.com/x6ySYEW.png

[iam-role-review]: https://i.imgur.com/PvoK0KA.png

[ec2-attach-iam-role]: https://i.imgur.com/cYGd3Kl.png

[ec2-attached-iam-role]: https://i.imgur.com/9SfgxAz.png

<!--TOC_START-->
## Contents
- [Overview](#overview)
- [Tutorial](#tutorial)
- [Exercises](#exercises)

<!--TOC_END-->
## Overview

> Identity Access Management (IAM) is a service which can allow the root user to create account, roles and groups. The root user also can grant permissions and to a certain degree of element. So, if a developer needs access to EC2 instance, rather than providing the credentials for the root user, the root user will create an account which has only access to EC2 instance. This is known as **Principle of least privilege**.

Reviewer: ```suggestion
**I**dentity **A**ccess **M**anagement (IAM) is a service which can allow the root user to create account, roles and groups. 
The root user also can grant permissions and to a certain degree of element. 
So, if a developer needs access to EC2 instance, rather than providing the credentials for the root user, the root user will create an account which has only access to EC2 instance. 
This is known as **Principle of least privilege**.
```

In this section you will be guided on how to create a role for EC2 instances to have access to RDS.

## Tutorial

1. Navigate to the AWS Console and sign in [here][aws-signin]

2. Search for IAM under the services drop-down menu.

![iam-service-search][iam-service-search]

> 3. You will then be redirected to the IAM Dashboard. You will need to go to the **Roles** section on the left panel and click **Create role** button.

Reviewer: ```suggestion
3. You will then be redirected to the IAM Dashboard. 
You will need to go to the **Roles** section on the left panel and click **Create role** button.
```

> 4. The next window will present you with the option to **Select type of trusted entity** and **Choose the service that will use this role**. By default, trusted entity should have checked **AWS Service**, do not change this. Select **EC2** as the service you will use this role. Proceed to Permissions.

Reviewer: ```suggestion
4. The next window will present you with the option to **Select type of trusted entity** and **Choose the service that will use this role**. 
By default, trusted entity should have checked **AWS Service**, do not change this. 
Select **EC2** as the service you will use this role. 
Proceed to Permissions.
```

![iam-role-create-role][iam-role-create-role]

> 5. Here, you will have the options to attach as many already made policies by Amazon or create your own custom AWS policy. Search **RDS** and check **AmazonRDSFullAccess**. This means that the role you are creating has all admin access to RDS services. Continue to the next section.

Reviewer: ```suggestion
5. Here, you will have the options to attach as many already made policies by Amazon or create your own custom AWS policy. 
Search **RDS** and check **AmazonRDSFullAccess**. 
This means that the role you are creating has all admin access to RDS services. 
Continue to the next section.
```

![iam-role-rds-policy][iam-role-rds-policy]

> 6. Tags are optional, but are useful for keeping track of what this role implies by its name. Proceed to reviewing your configuration.

Reviewer: ```suggestion
6. Tags are optional, but are useful for keeping track of what this role implies by its name. 
Proceed to reviewing your configuration.
```

![iam-role-tag][iam-role-tag]

> 7. You will need to add a name for your role. Once youre happy with this configuration, click **Create role**.

Reviewer: ```suggestion
7. You will need to add a name for your role. 
Once youre happy with this configuration, click **Create role**.
```

![iam-role-review][iam-role-review]

> 8. To use the role you have created, navigate back to your running [instances][running-instances] and select the instance you want to attach this role to. Go to **Actions** -> **Instance Settings** --> **Attach/Replace IAM Role**.

Reviewer: ```suggestion
8. To use the role you have created, navigate back to your running [instances][running-instances] and select the instance you want to attach this role to. 
Go to **Actions** -> **Instance Settings** --> **Attach/Replace IAM Role**.
```

![ec2-attach-iam-role][ec2-attach-iam-role]

> 9. Now you select the **Role** you want to attach. In this case, the role is: **Example-Role-For-EC2-Access-RDS**

Reviewer: ```suggestion
9. Now you select the **Role** you want to attach. 
In this case, the role is: **Example-Role-For-EC2-Access-RDS**
```
